### PR TITLE
Revert net48 prism change

### PR DESCRIPTION
**Pull request by pinzart90**
Issue:
After removing Microsoft.Practices.Prism for net48, DynamoRevit (revit 2024) stopped working in. Seems like the issue is that  Microsoft.Practices.Prism is missing.

Resolution:
Bring back Microsoft.Practices.Prism for net48
Keep Prism.Core for net6

### CODE REVIEW

File: src/DynamoCoreWpf/DynamoCoreWpf.csproj [hunk 1]
```diff
@@ -112,9 +112,6 @@
     <None Remove="Views\Menu\TrustedPathView.xaml" />
     <None Remove="Views\GuidedTour\HtmlPages\Resources\ConnectTheNode.gif" />
   </ItemGroup>
-  <ItemGroup Condition=" '$(TargetFramework)' == 'net6.0-windows' ">
-    <PackageReference Include="System.Configuration.ConfigurationManager" Version="5.0.0" />
-  </ItemGroup>
```
Comment (pinzart90) [inline src/DynamoCoreWpf/DynamoCoreWpf.csproj:117]: ConfigurationManager is included as part of net6-windows + wpf

File: src/DynamoCoreWpf/DynamoCoreWpf.csproj [hunk 2]
```diff
@@ -154,11 +151,13 @@
 	<PackageReference Include="SharpDX.Direct3D9" Version="4.2.0" />
 	<PackageReference Include="SharpDX.DXGI" Version="4.2.0" />
 	<PackageReference Include="SharpDX.Mathematics" Version="4.2.0" />
-    <Reference Include="Dynamo.Microsoft.Xaml.Behaviors, Version=1.0.0.0, Culture=neutral, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
+    <Reference Include="Dynamo.Microsoft.Xaml.Behaviors">
```
Comment (pinzart90) [inline src/DynamoCoreWpf/DynamoCoreWpf.csproj:154]: not specific version needed

File: src/DynamoUtilities/DynamoUtilities.csproj [hunk 3]
```diff
@@ -16,9 +16,7 @@
     <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
   </ItemGroup>
   <ItemGroup Condition=" '$(TargetFramework)' == 'net6.0' ">
-	  <PackageReference Include="System.Collections.Immutable" Version="5.0.0" />
     <PackageReference Include="System.Configuration.ConfigurationManager" Version="5.0.0" />
-    <PackageReference Include="System.IO.FileSystem.AccessControl" Version="5.0.0" />
```
Comment (pinzart90) [inline src/DynamoUtilities/DynamoUtilities.csproj:21]: these are already part of net6

File: src/GraphNodeManagerViewExtension/GraphNodeManagerViewExtension.csproj [hunk 4]
```diff
@@ -15,7 +15,6 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="FontAwesome.WPF" Version="4.7.0.9" />
-    <PackageReference Include="Prism.Core" Version="8.1.97" />
```
Comment (pinzart90) [inline src/GraphNodeManagerViewExtension/GraphNodeManagerViewExtension.csproj:18]: no need because of transitive references

File: src/DocumentationBrowserViewExtension/DocumentationBrowserViewExtension.csproj [hunk 5]
```diff
@@ -25,6 +25,9 @@
     <Reference Include="System.Windows.Forms" />
     <Reference Include="System.Xaml" />
     <Reference Include="WindowsBase" />
+    <Reference Include="Microsoft.Practices.Prism">
+        <HintPath>..\..\extern\prism\Microsoft.Practices.Prism.dll</HintPath>
```
Comment (mjkkirschner) [inline src/DocumentationBrowserViewExtension/DocumentationBrowserViewExtension.csproj:29]: should this be copy local false?

File: src/DynamoCoreWpf/Controls/ShortcutToolbar.xaml.cs [hunk 6]
```diff
@@ -6,7 +6,11 @@
 using Dynamo.UI.Commands;
 using Dynamo.ViewModels;
 using Dynamo.Wpf.ViewModels.Core;
-using Dynamo.Core;
+#if NETFRAMEWORK
+using NotificationObject = Microsoft.Practices.Prism.ViewModel.NotificationObject;
+#else
+using NotificationObject = Dynamo.Core.NotificationObject;
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Controls/ShortcutToolbar.xaml.cs:12]: so just curious... were we even using this type today in net48?

File: src/DynamoWPFCLI/DynamoWPFCLI.csproj [hunk 7]
```diff
@@ -12,12 +12,13 @@
...
   </PropertyGroup>
-  <ItemGroup>
-    <Reference Include="WindowsBase" />
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net48' ">
     <Reference Include="Microsoft.Practices.Prism">
-      <HintPath>..\..\extern\prism\Microsoft.Practices.Prism.dll</HintPath>
+        <HintPath>..\..\extern\prism\Microsoft.Practices.Prism.dll</HintPath>
     </Reference>
-    <Reference Include="WindowsBase" Condition=" '$(TargetFramework)' == 'net48' "/>
+    <Reference Include="WindowsBase" />
```
Comment (mjkkirschner) [inline src/DynamoWPFCLI/DynamoWPFCLI.csproj:19]: is this also included in net6-windows?

File: src/LibraryViewExtensionWebView2/LibraryViewExtensionWebView2.csproj [hunk 8]
```diff
@@ -192,6 +192,10 @@
     <Reference Include="System.Xml.Linq" />
     <Reference Include="System.Xaml" />
     <Reference Include="WindowsBase" />
+    <Reference Include="Microsoft.Practices.Prism">
+        <HintPath>..\..\extern\prism\Microsoft.Practices.Prism.dll</HintPath>
+        <Private>false</Private>
```
Comment (mjkkirschner) [inline src/LibraryViewExtensionWebView2/LibraryViewExtensionWebView2.csproj:197]: are both false and False valid?

File: src/Notifications/Notifications.csproj [hunk 9]
```diff
@@ -58,13 +58,16 @@
 	<ItemGroup>
 		<PackageReference Include="FontAwesome.WPF" Version="4.7.0.9" />
 		<PackageReference Include="Microsoft.Web.WebView2" Version="1.0.1264.42" />
-        <PackageReference Include="Prism.Core" Version="8.1.97" />
```
Comment (mjkkirschner) [inline src/Notifications/Notifications.csproj:61]: transitive?

File: test/Engine/ProtoTest/ProtoTest.csproj [hunk 10]
```diff
@@ -36,6 +36,7 @@
           <Reference Include="System.Xml" />
           <Reference Include="EmbeddedInterop" Condition=" '$(TargetFramework)' == 'net48'">
               <Private>False</Private>
+              <HintPath>..\..\test_support_projects\EmbeddedInterop.dll</HintPath>
```
Comment (mjkkirschner) [inline test/Engine/ProtoTest/ProtoTest.csproj:39]: unrelated?

**Review by mjkkirschner — APPROVED** [review]
a few questions

File: src/DocumentationBrowserViewExtension/DocumentationBrowserViewExtension.csproj (hunk 5 above)
Comment (pinzart90) [reply src/DocumentationBrowserViewExtension/DocumentationBrowserViewExtension.csproj:29]: it was not set before. I think it could be
I just restored it as it was

File: src/DynamoCoreWpf/Controls/ShortcutToolbar.xaml.cs (hunk 6 above)
Comment (pinzart90) [reply src/DynamoCoreWpf/Controls/ShortcutToolbar.xaml.cs:12]: NotificationObject is used in net48
see here https://github.com/DynamoDS/Dynamo/blob/0323558faa1634097e3d3cae0e2ad0cf48cd3bac/src/DynamoCoreWpf/Controls/ShortcutToolbar.xaml.cs#L232

File: src/DynamoWPFCLI/DynamoWPFCLI.csproj (hunk 7 above)
Comment (pinzart90) [reply src/DynamoWPFCLI/DynamoWPFCLI.csproj:19]: Yes it is.

File: src/DynamoCoreWpf/Controls/ShortcutToolbar.xaml.cs (hunk 6 above)
Comment (mjkkirschner) [reply src/DynamoCoreWpf/Controls/ShortcutToolbar.xaml.cs:12]: sorry @pinzart90 I should be more clear. Are we using `Dynamo.Core.NotificationObject` in net48? Or is that type you linked a subclass of the Microsoft Type? Or am I misunderstanding and that is a new type only in the new prism?

File: src/LibraryViewExtensionWebView2/LibraryViewExtensionWebView2.csproj (hunk 8 above)
Comment (pinzart90) [reply src/LibraryViewExtensionWebView2/LibraryViewExtensionWebView2.csproj:197]: yes

File: test/Engine/ProtoTest/ProtoTest.csproj (hunk 10 above)
Comment (pinzart90) [reply test/Engine/ProtoTest/ProtoTest.csproj:39]: unrelated. Caused warnings

File: src/DynamoCoreWpf/Controls/ShortcutToolbar.xaml.cs (hunk 6 above)
Comment (pinzart90) [reply src/DynamoCoreWpf/Controls/ShortcutToolbar.xaml.cs:12]: Dynamo.Core.NotificationObject is used in net48. 
Dynamo.Core.NotificationObject and Microsoft.Practices.Prism.ViewModel.NotificationObject are separate implementations and they serve the same purpose

**Review by sm6srw — APPROVED** [review]

File: src/DynamoCoreWpf/Controls/ShortcutToolbar.xaml.cs (hunk 6 above)
Comment (aparajit-pratap) [reply src/DynamoCoreWpf/Controls/ShortcutToolbar.xaml.cs:12]: Still confused. When is which used in net48? Do we need both? What about in net6? Do we only need Dynamo.Core.NotificationObject in net6?

File: src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerViewModel.cs [hunk 11]
```diff
@@ -1,5 +1,9 @@
 using Dynamo.ViewModels;
-using Dynamo.Core;
+#if NETFRAMEWORK
+using NotificationObject = Microsoft.Practices.Prism.ViewModel.NotificationObject;
+#else
+using NotificationObject = Dynamo.Core.NotificationObject;
```
Comment (aparajit-pratap) [inline src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerViewModel.cs:5]: Can you explain again why `Dynamo.Core.NotificationObject` doesn't work in .net48?

File: src/DynamoCoreWpf/Controls/ShortcutToolbar.xaml.cs (hunk 6 above)
Comment (mjkkirschner) [reply src/DynamoCoreWpf/Controls/ShortcutToolbar.xaml.cs:12]: yeah, also still confused. If Dynamo.Core.NotificationObject is used in net48 - why are you doing this:
```
#if NETFRAMEWORK
using NotificationObject = Microsoft.Practices.Prism.ViewModel.NotificationObject;
```

File: src/Libraries/PythonNodeModelsWpf/PythonNodeModelsWpf.csproj [hunk 12]
```diff
@@ -42,29 +42,21 @@
...
     <Reference Include="PresentationCore">
-      <Private>False</Private>
+        <Private>False</Private>
     </Reference>
     <Reference Include="PresentationFramework">
-      <Private>False</Private>
+        <Private>False</Private>
     </Reference>
     <Reference Include="System.Xaml" />
     <Reference Include="WindowsBase" />
```
Comment (aparajit-pratap) [inline src/Libraries/PythonNodeModelsWpf/PythonNodeModelsWpf.csproj:56]: Isn't this required for net48?

File: src/DynamoCoreWpf/Controls/ShortcutToolbar.xaml.cs (hunk 6 above)
Comment (pinzart90) [reply src/DynamoCoreWpf/Controls/ShortcutToolbar.xaml.cs:12]: in net48:
Both Microsoft.Practices.Prism.ViewModel.NotificationObject and Dynamo.Core.NotificationObject are used. 
in net6:
NotificationObject is no longer part of Prism.Core. We switched to Dynamo.Core.NotificationObject

I wanted to revert all changes in net48 because some of the Microsoft.Practices.Prism types (including NotificationObject) bubble up to the public API. We cannot be sure how these types are used so better safe than sorry.
For example dynamo revit uses some of these types and expects that dynamo will have the Microsoft.Practices.Prism.dll

File: src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerViewModel.cs (hunk 11 above)
Comment (pinzart90) [reply src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerViewModel.cs:5]: replacing all Microsoft.Practices.Prism.ViewModel.NotificationObject with Dynamo.Core.NotificationObject might cause breaking changes in the public API.
From the little testing I did in DynamoRevit it looked fine, but I cannot be sure about other Dynamo API use cases.

File: src/Libraries/PythonNodeModelsWpf/PythonNodeModelsWpf.csproj (hunk 12 above)
Comment (pinzart90) [reply src/Libraries/PythonNodeModelsWpf/PythonNodeModelsWpf.csproj:56]: Yes, it probably is required by net48
I just changed the spacing
